llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -10348,6 +10348,7 @@ SDValue RISCVTargetLowering::LowerINTRINSIC_WO_CHAIN(SDValue Op,
     return DAG.getNode(RISCVISD::MOPRR, DL, XLenVT, Op.getOperand(1),
                        Op.getOperand(2), Op.getOperand(3));
   }
+  case Intrinsic::clmul:
```

Comment:
Target independent intrinsics should be converted to a `ISD::CLMUL` in SelectionDAGBuilder.cpp

Reply:
now that this is in SelectionDAGBuilder should it be fully removed from here? If so, how do I tell LLVM to turn `llvm.clmul` into a RiscV `clmul` assembly instruction?

Reply:
is it that I keep the change on 14288 of this file, but drop this one?

Reply:
We should not do that. We should add `clmul` to TargetSelectionDAG.td and add isel patterns to RISCVInstrInfoZb.td like

```
def : PatGprGpr<clmul, CLMUL>;
```

Reply:
what is `PatGprGpr`?

Reply:
It's a class defined in RISCVInstrInfo.td. It's an isel pattern that takes 2 register inputs and produces 1 register result.